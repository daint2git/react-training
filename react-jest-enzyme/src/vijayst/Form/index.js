import React from 'react'

export default class Form extends React.Component {
  state = { email: '' }

  handleMailChange = e => {
    this.setState({ email: e.target.value })
  }

  handleAdd = e => {
    e.preventDefault()
    this.props.onAdd(this.state.email)
  }

  render() {
    return (
      <form>
        <input
          type="text"
          name="email"
          value={this.state.value}
          onChange={this.handleMailChange}
        />
        <button onClick={this.handleAdd}>Add</button>
      </form>
    )
  }
}
