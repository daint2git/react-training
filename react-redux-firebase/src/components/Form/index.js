import React, { useState } from 'react'
import { connect } from 'react-redux'

import { createTodo } from 'redux/reducers/todos'
import File from 'components/File'
import Animation from 'components/Animation'

const Form = ({ createTodo }) => {
  const [value, updateValue] = useState('')
  return (
    <div>
      <h3>Todo Form</h3>
      <form
        onSubmit={e => {
          e.preventDefault()
          createTodo({ task: value })
          updateValue('')
        }}
      >
        <input type="text" value={value} onChange={e => updateValue(e.target.value)} />
        <File />
        <Animation />
        <input type="submit" />
      </form>
    </div>
  )
}

const mapDispatchToProps = {
  createTodo,
}

export default connect(
  null,
  mapDispatchToProps,
)(Form)
