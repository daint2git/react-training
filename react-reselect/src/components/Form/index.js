import React, { useState } from 'react'

const Form = ({ onSubmit }) => {
  console.log('render Form')
  const [value, updateValue] = useState('')
  return (
    <div>
      <h3>Todo Form</h3>
      <form
        onSubmit={e => {
          e.preventDefault()
          onSubmit({
            task: value,
          })
          updateValue('')
        }}
      >
        <input type="text" value={value} onChange={e => updateValue(e.target.value)} />
        <input type="submit" />
      </form>
    </div>
  )
}

export default React.memo(Form)
