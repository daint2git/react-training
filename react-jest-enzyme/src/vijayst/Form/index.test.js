import { mount, shallow, render } from 'enzyme'
import renderer from 'react-test-renderer'
import Form from './'

describe('Form', () => {
  describe('Snapshot', () => {
    test('1. renderer', () => {
      const component = renderer.create(<Form />)
      const json = component.toJSON()
      expect(json).toMatchSnapshot()
    })

    test('2. shallow', () => {
      const wrapper = shallow(<Form />)
      expect(wrapper).toMatchSnapshot()
    })

    test('3. mount', () => {
      const wrapper = mount(<Form />)
      expect(wrapper).toMatchSnapshot()
    })

    test('4. render', () => {
      const wrapper = render(<Form />)
      expect(wrapper).toMatchSnapshot()
    })
  })

  describe('handles', () => {
    let wrapper, onAdd

    beforeEach(() => {
      onAdd = jest.fn()
      wrapper = mount(<Form onAdd={onAdd} />)
    })

    afterEach(() => {
      wrapper.unmount()
    })

    test('Form requires onAdd prop', () => {
      expect(wrapper.props().onAdd).toBeDefined()
    })

    test('Form renders button', () => {
      const button = wrapper.find('button').first()
      expect(button).toBeDefined()
    })

    test('Button click calls onAdd', () => {
      const button = wrapper.find('button').first()
      const input = wrapper.find('input').first()

      input.simulate('change', {
        target: { value: 'example@email' },
      })
      button.simulate('click')

      expect(onAdd).toBeCalledWith('example@email')
    })
  })
})
