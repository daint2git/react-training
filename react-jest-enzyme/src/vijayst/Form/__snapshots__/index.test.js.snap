// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Form Snapshot 1. renderer 1`] = `
<form>
  <input
    name="email"
    onChange={[Function]}
    type="text"
  />
  <button
    onClick={[Function]}
  >
    Add
  </button>
</form>
`;

exports[`Form Snapshot 2. shallow 1`] = `
<form>
  <input
    name="email"
    onChange={[Function]}
    type="text"
  />
  <button
    onClick={[Function]}
  >
    Add
  </button>
</form>
`;

exports[`Form Snapshot 3. mount 1`] = `
<form>
  <input
    name="email"
    onChange={[Function]}
    type="text"
  />
  <button
    onClick={[Function]}
  >
    Add
  </button>
</form>
`;

exports[`Form Snapshot 4. render 1`] = `
<form>
  <input
    name="email"
    type="text"
  />
  <button>
    Add
  </button>
</form>
`;
